src/ucp/wireup/address.c

```diff
@@ -664,23 +670,38 @@ static ucs_status_t ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep,
                     ucp_address_memcheck(context, ptr, ep_addr_len, rsc_index);
                     ptr = UCS_PTR_BYTE_OFFSET(ptr, ep_addr_len);
 
+                    /* pack ep lane index */
+                    remote_lane    = (lanes2remote == NULL) ? lane :
+                                     lanes2remote[lane];
+                    *(uint8_t*)ptr = remote_lane;
+                    ptr            = UCS_PTR_TYPE_OFFSET(ptr, uint8_t);
```

Comment:
can use `UCS_PTR_TYPE_OFFSET(ptr, remote_lane)`

Reply:
https://github.com/openucx/ucx/pull/4332#discussion_r341019287 - we assigned a value to ptr casted to uint8_t*, so need to advance by sizeof(uint8_t)